### DISCUSSION TITLE
主备自动切换模式部署中端口的值是怎么来的？如果要修改怎么修改？

### DISCUSSION THREAD

**Discussion by heshiyingx** [apache/rocketmq#6856] [Q&A] [answered]
在官方文档中给的 NameServer 的配置文件部署示例
```
enableControllerInNamesrv = true
controllerDLegerGroup = group1
controllerDLegerPeers = n0-127.0.0.1:9877;n1-127.0.0.1:9878;n2-127.0.0.1:9879
controllerDLegerSelfId = n0
controllerStorePath = /home/admin/DledgerController
enableElectUncleanMaster = false
notifyBrokerRoleChanged = true
```
请问controllerDLegerPeers(DLedger Group 内各节点的端口信息)中那三个端口是怎么来的(9877,9878,9879)?对应节点中哪个配置项？
谢谢各位大佬！麻烦各位大佬指导一下

**Comment by TheR1sing3un** [comment]
这个端口是controller组件和broker交互的端口，如果你需要改的话，直接在`controllerDLedgerPeers` 中修改后面的peers的端口就可以

**Reply by heshiyingx** [reply]
麻烦再请教一下，这三个端口用于controller组件和broker交互。应该是controller去访问broker吧，就是说某个broker应该需要监听这三个端口中的一个。怎么在broker上去配置这个交互所使用的端口呢？

**Reply by TheR1sing3un** [reply, marked as answer]
> 麻烦再请教一下，这三个端口用于controller组件和broker交互。应该是controller去访问broker吧，就是说某个broker应该需要监听这三个端口中的一个。怎么在broker上去配置这个交互所使用的端口呢？

这个是broker访问controller的端口，比如说broker去controller注册，controller访问broker的话，是通过broker往controller注册的信息里面有broker的地址。

如果你修改了controller部署的配置文件的地址/端口这些的话，broker的配置文件也需要改。

<img width="820" alt="image" src="https://github.com/apache/rocketmq/assets/87409330/8b4301d2-50b1-499a-af34-3066e314f844"> 

参考这个配置参数`controllerAddr`，后面填你的controller的ip:port即可，多个controller就填多个

**Reply by heshiyingx** [reply]
非常感谢您的指导，谢谢
